bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/builders/ShadeChannelBuilder.java

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hdpowerview.internal.builders;
+
+import java.util.function.Predicate;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.hdpowerview.internal.HDPowerViewTranslationProvider;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.type.ChannelKind;
+import org.openhab.core.thing.type.ChannelTypeUID;
+
+/**
+ * The {@link ShadeChannelBuilder} class creates vane and secondary position channels.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+public class ShadeChannelBuilder {
```

Comment:
Maybe I mislead you, but when suggesting to provide a channel builder, intention was to provide another builder following the same pattern as the existing builders. Existing builders:
- Each builder builds only one channel.
- Each builder is a subclass of `BaseChannelBuilder`.
- Each builder contains a static method `create` returning a builder instance.
- Builders are used by things but are completely decoupled from things (i.e. contain no references to handlers).
- With all of the above, these builders are quite focused and simple (`SceneChannelBuilder` being the simplest and `AutomationChannelBuilder` containing most logic).

I'll let you consider this before starting a review of the current implementation.

Reply:
Your idea was basically excellent because with a builder one can use the **_exact same algorithm_** for creating and adding resp. removing each of the three possible dynamic things.

I looked at your BaseChannelBuilder, HOWEVER I could not use it because your builder is designed to create MULTIPLE instances of the SAME channel type, hosted in a BRIDGE, and WITH the channels in a channel group. Whereas in this case we need to create SINGLE instances of MULTIPLE channel types, hosted in THING, and WITHOUT a channel group.

In short it was easier to create a dedicated new builder class than to 'munge' (aka re-purpose) your class.

To respond to your comments ..
- Each builder builds only one channel => yes
- Each builder is a subclass of BaseChannelBuilder => no (see above)
- Each builder contains a static method create returning a builder instance => Huh? I honestly don't see any difference between calling 'new construstor()' versus calling static create() which calls 'new construstor()' ..
=> Builders are used by things but are completely decoupled from things (i.e. contain no references to handlers) => yes
- .. builders are quite focused and simple => yes (and my class is not extended from BaseChannelBuilder..)